src/Core/Containers/VariableSet.hpp

```diff
@@ -359,35 +434,64 @@ class RA_CORE_API VariableSet
 
     /// The core of a container logic : a template member variable
     template <typename T>
-    static std::unordered_map<const VariableSet*, BaseContainer<T>> m_variables;
+    static std::unordered_map<const VariableSet*, VariableContainer<T>> m_variables;
 
-    /// \brief test if the type F has a callable operator with the right profile
+    /// Implementation of static visitor
+    /// \{
+    /// \brief Callable profile of F for variable of type T
     template <typename F, typename T>
     using VisitFunction = decltype(
         std::declval<F>().operator()( std::declval<const std::string&>(), std::declval<T&>() ) );
 
-    /// Implementation of static visitor
-    /// \{
+    /// \brief Callable profile of F for variable of type T and user parameter of type U&&
+    template <typename F, typename T, typename U>
+    using VisitFunctionUP =
+        decltype( std::declval<F>().operator()( std::declval<const std::string&>(),
+                                                std::declval<T&>(),
+                                                std::declval<U&&>() ) );
+
+    /// \brief test if the type F has a callable operator with the right profile
     template <typename F, typename T>
     static constexpr bool has_visit_v = Ra::Core::Utils::is_detected<VisitFunction, F, T>::value;
 
+    /// \brief test if the type F has a callable operator with the right profile
+    template <typename F, typename T, typename U>
+    static constexpr bool has_visit_up_v =
+        Ra::Core::Utils::is_detected<VisitFunctionUP, F, T, U>::value;
+
+    /// unfold type list to generate visitor func call
     template <typename F, template <typename...> typename TYPESLIST, typename... TYPES>
     void visitImpl( F&& visitor, TYPESLIST<TYPES...> ) const;
 
     template <typename F, typename T>
     void visitImplHelper( F& visitor ) const;
+
+    /// unfold type list to generate visitor func call
+    template <typename F, typename U, template <typename...> typename TYPESLIST, typename... TYPES>
+    void visitImplUp( F&& visitor, U&& userParam, TYPESLIST<TYPES...> ) const;
```

Comment:
Up has upward or UserParameter ? 
I'm not sure I catch the naming of this function

Reply:
UserParameter.
I'll try to find better naming asap.